### PR TITLE
Add additional quadrature rule for Simplex

### PR DESCRIPTION
**Pull request by peterrum**
I am not sure about the name of the class...

### CODE REVIEW

File: include/deal.II/simplex/quadrature_lib.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+  /**
+   * Integration rule for simplex entities.
+   *
+   * Following number of quadrature points are currently supported:
+   *   - 2D: 1, 3, 7
+   *   - 3D: 1, 4, 10
+   */
+  template <int dim>
+  class PGauss : public Quadrature<dim>
```
Comment (luca-heltai) [inline include/deal.II/simplex/quadrature_lib.h:36]: Could you derive this from QSimplex?

**Comment by luca-heltai** [comment]
> I am not sure about the name of the class...

I think the name is just fine. :)

File: include/deal.II/simplex/quadrature_lib.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/simplex/quadrature_lib.h]: Done!

File: include/deal.II/simplex/quadrature_lib.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Constructor taking the number of quadrature points @p n_points.
+     */
+    PGauss(const unsigned int n_points);
+  };
+} // namespace Simplex
+
+DEAL_II_NAMESPACE_CLOSE
+
+#endif
```
Comment (tjhei) [inline include/deal.II/simplex/quadrature_lib.h:48]: missing newline

File: source/simplex/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+SET(_inst
+)
+
+FILE(GLOB _header
+  ${CMAKE_SOURCE_DIR}/include/deal.II/tet/*.h
+  )
+
+DEAL_II_ADD_LIBRARY(obj_tet OBJECT ${_src} ${_header} ${_inst})
+EXPAND_INSTANTIATIONS(obj_tet "${_inst}")
```
Comment (tjhei) [inline source/simplex/CMakeLists.txt:42]: missing newline

File: source/simplex/quadrature_lib.cc [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+           ExcMessage("No valid quadrature points!"));
+  }
+
+} // namespace Simplex
+
+
+template class Simplex::PGauss<2>;
+template class Simplex::PGauss<3>;
+
+DEAL_II_NAMESPACE_CLOSE
```
Comment (tjhei) [inline source/simplex/quadrature_lib.cc:146]: missing newline

File: source/simplex/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  )
+
+SET(_inst
+)
+
+FILE(GLOB _header
+  ${CMAKE_SOURCE_DIR}/include/deal.II/tet/*.h
+  )
+
+DEAL_II_ADD_LIBRARY(obj_tet OBJECT ${_src} ${_header} ${_inst})
```
Comment (tjhei) [inline source/simplex/CMakeLists.txt:41]: should we name it obj_simplex instead?

File: include/deal.II/simplex/quadrature_lib.h [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+#define dealii_tet_quadrature_lib_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/quadrature_lib.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Simplex
```
Comment (tjhei) [inline include/deal.II/simplex/quadrature_lib.h:26]: namespace documentation? Otherwise it won't show up in doxygen.

Comment (peterrum) [reply include/deal.II/simplex/quadrature_lib.h:26]: One of the other PRs contains the namespace docu!

**Comment by peterrum** [comment]
@tjhei I have made the changes!

File: include/deal.II/simplex/quadrature_lib.h [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+   *   - 3D: 1, 4, 10
+   */
+  template <int dim>
+  class PGauss : public QSimplex<dim>
+  {
+  public:
+    /**
+     * Constructor taking the number of quadrature points @p n_points.
+     */
+    PGauss(const unsigned int n_points);
```
Comment (tjhei) [inline include/deal.II/simplex/quadrature_lib.h:42]: mark explicit

Comment (peterrum) [reply include/deal.II/simplex/quadrature_lib.h]: Done!

**Comment by tjhei** [comment]
Is this supposed to be guarded by WITH_SIMPLEX_SUPPORT? An additional advantage would be that it won't show up in doxygen for now.

**Comment by peterrum** [comment]
> Is this supposed to be guarded by WITH_SIMPLEX_SUPPORT? An additional advantage would be that it won't show up in doxygen for now.

Personally, I would like to keep the usage of `WITH_SIMPLEX_SUPPORT` to a minimal and use it only at a few critical places. Or what is your opinion?

**Comment by tjhei** [comment]
> Or what is your opinion?

Then I misremember the reason for adding the WITH_SIMPLEX flag. I thought it would be a way to hide away everything related to TET support (from using and from doxygen) until we agree that it is ready .

**Comment by luca-heltai** [comment]
This is also useful for non-simplex related things, like `QSplit` (used in BEM with quadrilaterals).

**Pull request merged**